### DISCUSSION TITLE
A 'SplitCHParsed' is not a registered object.

### DISCUSSION THREAD

**Discussion by FatimahDzr** [idaholab/moose#21946] [Q&A Modules: General] [answered]
Hello everyone,

I tried to run the code [2015_10_19_FAST_2p_periodic_neckgrowth.i](https://github.com/SudiptaBiswas/Crow/blob/verified_final_codes/tests/2015_10_Sintering/2015_10_19_FAST_2p_periodic_neckgrowth.i) from [SudiptaBiswas](https://github.com/SudiptaBiswas/Crow/commits?author=SudiptaBiswas). Even though Im connected with her directory and linked my moose to GitHub, every time I tried to run the code via UBUNTU and using  ../projects/crow/crow-opt -i 2017_04_06_2p_matrxmob.i command I get this error:

*** ERROR ***
**A 'SplitCHParsed' is not a registered object.**

If you are trying to find this object in a dynamically linked library, make sure that
the library can be found either in your "Problem/library_path" parameter or in the
MOOSE_LIBRARY_PATH environment variable.

Stack frames: 22
0: libMesh::print_trace(std::ostream&)
1: moose::internal::mooseErrorRaw(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)
2: void mooseError<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&&)
3: Factory::reportUnregisteredError(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const
4: Factory::getValidParams(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
5: MooseObjectAction::MooseObjectAction(InputParameters)
6: AddKernelAction::AddKernelAction(InputParameters)
7: /mnt/c/For_MOOSE/projects/moose/framework/libmoose-opt.so.0(+0x795588) [0x7fbfd3a39588]
8: /mnt/c/For_MOOSE/projects/moose/framework/libmoose-opt.so.0(+0x795617) [0x7fbfd3a39617]
9: ActionFactory::create(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, InputParameters&)
10: Parser::walkRaw(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, hit::Node*)
11: Parser::walk(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, hit::Node*)
12: hit::Node::walk(hit::Walker*, hit::NodeType, hit::TraversalOrder)
13: hit::Node::walk(hit::Walker*, hit::NodeType, hit::TraversalOrder)
14: hit::Node::walk(hit::Walker*, hit::NodeType, hit::TraversalOrder)
15: Parser::parse(std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&)
16: MooseApp::setupOptions()
17: MooseApp::run()
18: main
19: /lib/x86_64-linux-gnu/libc.so.6(+0x29d90) [0x7fbfcc647d90]
20: __libc_start_main
21: ../projects/crow/crow-opt(+0x23ef) [0x5645364fd3ef]
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0
[unset]: write_line error; fd=-1 buf=:cmd=abort exitcode=1
:
system msg for write_line failure : Bad file descriptor

As I understand the kernel A 'SplitCHParsed' is not a registered in my Moose library. I would be grateful if you could help me with this error and how to solve that. Is there any approach to adding a kernel to the moose library?

**Comment by GiudGiud** [comment, marked as answer]
Hello

so you are using Crow directly? 
you’re not hitting any error? There were issues like 2 months ago did they get addressed? 

can you please paste the output ´ ldd crow-opt’
To see if the phase field module for linked properly 

Guillaume

**Reply by GiudGiud** [reply]
Also can you more simple input files with crow?

**Reply by FatimahDzr** [reply]
I can run the simple examples in the moose example file but for this case, it cannot be run. I downloaded SplitCHParsed kernel and pasted it into files "include" and "src" to address this issue. It worked and apparently the code can be run but when it reached the kernel "PolycrystalSinteringKernel", it stopped and printed 
*** ERROR ***
/mnt/c/moose/exp/ex10.i:81: missing required parameter 'Kernels/PolycrystalSinteringKernel/type'
        Doc String: "A string representing the Moose Object that will be built by this Action"

I couldn't find that kernel yet. I'm not sure if it is a practical solution to download and put each kernel for running the problem.

**Reply by GiudGiud** [reply]
you forgot to specify the type of the object, it s a required parameter 
otherwise we dont know what the object is

**Reply by FatimahDzr** [reply]
Thank you for your response, at first I tried to create my own application but every time an error appeared. So I used cloning to the Crow git. I cloned to the "devel" branch successfully but when I'm going to create crow-opt, this error appears :
```
/home/fatimahdzr/mambaforge3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /mnt/c/moose/projects/gav/lib/libCrow-opt.so: **undefined reference to `InputParameters validParams<Material>()'
collect2: error: ld returned 1 exit status**
```
I would be grateful if you could help me with this error.
Regards

**Reply by GiudGiud** [reply]
Hello

This is because Crow has deprecated syntax for valid parameters. see this post
https://mooseframework.inl.gov/newsletter/2020_04.html#!

We really want someone to fix this, it keeps coming up

Guillaume
